app/services/prison_offender_manager_service.rb

```diff
@@ -2,17 +2,24 @@ class PrisonOffenderManagerService
   def self.get_pom_detail(nomis_staff_id)
     PomDetail.find_or_create_by!(nomis_staff_id: nomis_staff_id.to_i) { |s|
       s.working_pattern = s.working_pattern || 0.0
-      s.status = s.status || 'inactive'
+      s.status = s.status || 'active'
     }
   end
 
-  def self.get_poms(prison)
+  def self.get_poms(prison, &filter)
     poms = Nomis::Elite2::Api.prisoner_offender_manager_list(prison)
-    poms.map { |pom|
+
+    poms = poms.map { |pom|
       detail = get_pom_detail(pom.staff_id)
       pom.add_detail(detail)
       pom
-    }
+    }.compact
+
+    if filter
```

Comment:
could also do `if block_given?` if you want to do away with `&filter` param

Reply:
Wanted to name it to be explicit, so that you've an idea of what it does.  block_given? could be anything and I'm not sure it'd be at all obvious.  

Happy to change it though if you think it's more idiomatic though @TobyRet